lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -0,0 +1,745 @@
+
+
+class ScalewayComputeAPI(object):
+
+    def __init__(self, module, headers, base_url):
+        self.module = module
+        self.headers = headers
+        self.base_url = base_url
+
+    def _url_builder(self, path):
+        if path[0] == '/':
+            path = path[1:]
+        return '%s/%s' % (self.base_url, path)
+
+    def send(self, method, path, data=None, headers=None):
+        url = self._url_builder(path)
+        data = self.module.jsonify(data)
+        timeout = self.module.params['timeout']
+
+        if headers is not None:
+            self.headers.update(headers)
+
+        resp, info = fetch_url(self.module, url, data=data, headers=self.headers, method=method, timeout=timeout)
+
+        # Exceptions in fetch_url may result in a status -1, the ensures a proper error to the user in all cases
+        if info['status'] == -1:
+            self.module.fail_json(msg=info['msg'])
+
+        return Response(resp, info)
+
+    def get(self, path, data=None, headers=None):
+        return self.send('GET', path, data, headers)
+
+    def put(self, path, data=None, headers=None):
+        return self.send('PUT', path, data, headers)
+
+    def post(self, path, data=None, headers=None):
+        return self.send('POST', path, data, headers)
+
+    def delete(self, path, data=None, headers=None):
+        return self.send('DELETE', path, data, headers)
+
+    def patch(self, path, data=None, headers=None):
+        return self.send("PATCH", path, data, headers)
+
+    def update(self, path, data=None, headers=None):
+        return self.send("UPDATE", path, data, headers)
+
+
+def is_present(compute_api, server):
+    print_warning(compute_api, "We test to see if the server %s it's present or not" % server["id"])
+    return True
+
+
+def create_server(compute_api, server):
+    print_warning(compute_api, "Starting a create_server")
+    target_server = None
+    response = compute_api.post(path="servers",
+                                data={"enable_ipv6": server["enable_ipv6"],
+                                      "tags": server["tags"],
+                                      "commercial_type": server["commercial_type"],
+                                      "image": server["image"],
+                                      "name": server["name"],
+                                      "organization": server["organization"]})
+
+    if not response.ok:
+        msg = 'Error during server creation: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    try:
+        target_server = response.json["server"]
+    except KeyError:
+        compute_api.module.fail_json(msg="Error in getting the server information from: %s" % response.json)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(create_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=target_server)
```

Comment:
test of `wait` parameter could be moved inside the `wait_to_complete_state_transition` method.